### DISCUSSION TITLE
blocksize function, is it possible to limit the video output size

### DISCUSSION THREAD

**Discussion by sugizo** [python-ffmpegio/python-ffmpegio#39] [Q&A] [answered]
have 2 questions : 
1. what is blocksize parameter in open() function ?
2. how to limit video output size (in mb, or gb) ?

**code**
```
input = 'http://195.196.36.242/mjpg/video.mjpg'
output = 'rtsp.mp4'
with ffmpegio.open(input, 'rv') as fin, ffmpegio.open(output, 'wv', rate_in = fin.rate, blocksize = 1, overwrite = True) as fout:
    for frames in fin:
        fout.write(frames)
```

**result**
no error occured, but seems looping forever

**expected result**
the loop is ended when the output file 'rtsp.mp4' size is 1 mb

best regards

**Comment by tikuma-lsuhsc** [comment, marked as answer]
> what is blocksize parameter in open() function ?

sets the number of frames to read at once (only used by a reader or filter streams). For example,

```python
with ffmpegio.open(input, 'rv', blocksize=100) as fin
   for frames in fin:
        assert frames.shape[0]==100
```

should read all the frames. The last block likely does not contain 100 frames, and it would results in the assertion error.

> how to limit video output size (in mb, or gb) ?

This is a quintessential FFmpeg question. The short answer is you cannot in a one-pass encoding. You can use a two-pass encoding to specify the size. See https://trac.ffmpeg.org/wiki/Encode/H.264#twopass

However, `ffmpegio` currently provide a simple interface for `transcode` function (See the example on [the readme page](https://github.com/python-ffmpegio/python-ffmpegio?tab=readme-ov-file#transcoding). Actually, I'm not even sure if you can do it with an input pipe, which is requirement for with any `ffmpegio` writer function/class to do two-pass operation.

But I don't think this is what you're interested in. You want to know if you can stop the reader stream when it has written so many frames. As far as I know, you cannot do it by the file size, but you can specify the duration in seconds by `s` option or in frames by `vframes` option. See the [FFmpeg documentation](https://ffmpeg.org/ffmpeg.html#Main-options). 

There is a way to terminate it by the number of bytes read using the progress callback. But I'm spotting bugs which freezes the execution (interlocked thread). If you're interested, I'll remember to come back to this thread for a solution once the bugs are sorted out.

**Comment by sugizo** [comment]
nice thx
```
output = 'rtsp_ffmpeg.mp4'
ffmpeg("-y -i 'http://195.196.36.242/mjpg/video.mjpg' -frames 9 -c libx264 %s" % output)
```
